src/main/java/dev/aura/bungeechat/command/MessageToggleCommand.java

```diff
@@ -22,17 +27,44 @@ public MessageToggleCommand(MessengerModule messengerModule) {
   public void execute(CommandSender sender, String[] args) {
     if (!PermissionManager.hasPermission(sender, Permission.COMMAND_TOGGLE_MESSAGE)) return;
 
-    if (!(sender instanceof ProxiedPlayer)) {
-      MessagesService.sendMessage(sender, Messages.NOT_A_PLAYER.get());
-    } else {
-      BungeeChatAccount player = BungeecordAccountManager.getAccount(sender).get();
-      player.toggleMessanger();
+    if (args.length == 0) {
+      if (!(sender instanceof ProxiedPlayer)) {
+        MessagesService.sendMessage(sender, Messages.NOT_A_PLAYER.get());
+      } else {
+        BungeeChatAccount player = BungeecordAccountManager.getAccount(sender).get();
+        player.toggleMessanger();
 
+        if (player.hasMessangerEnabled()) {
+          MessagesService.sendMessage(sender, Messages.ENABLE_MESSAGER.get());
+        } else {
+          MessagesService.sendMessage(sender, Messages.DISABLE_MESSAGER.get());
+        }
+      }
+    } else if (args.length == 1) {
+      if (!PermissionManager.hasPermission(sender, Permission.COMMAND_TOGGLE_MESSAGE_OTHERS))
+        return;
+      BungeeChatAccount player = BungeecordAccountManager.getAccount(args[0]).get();
+      player.toggleMessanger();
       if (player.hasMessangerEnabled()) {
-        MessagesService.sendMessage(sender, Messages.ENABLE_MESSAGER.get());
+        MessagesService.sendMessage(sender, Messages.ENABLE_MESSAGER_OTHERS.get(player));
       } else {
-        MessagesService.sendMessage(sender, Messages.DISABLE_MESSAGER.get());
+        MessagesService.sendMessage(sender, Messages.DISABLE_MESSAGER_OTHERS.get(player));
       }
+    } else {
+      MessagesService.sendMessage(
+          sender, Messages.INCORRECT_USAGE.get(sender, "/msgtoggle [player]"));
+    }
+  }
+
+  @Override
+  public Collection<String> tabComplete(CommandSender sender, String[] args) {
+    if (args.length == 1) {
+      return BungeecordAccountManager.getAccountsForPartialName(args[0], sender).stream()
+              .filter(account -> account.getAccountType() == AccountType.PLAYER)
+              .map(BungeeChatAccount::getName)
```

Comment:
Also filter out the player themselves. Use the filter used in the linked code snipped in addition: https://github.com/AuraDevelopmentTeam/BungeeChat2/blob/b98a77557e905cc9fb99755a6f3e3960b9d2f364/src/main/java/dev/aura/bungeechat/command/MessageCommand.java#L64-L69